src/components/Dashboard/Analytics/AnalyticsMainBarChart.jsx

```diff
@@ -1,18 +1,33 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { Bar } from "react-chartjs-2";
 import { BarChartContainer } from "./AnalyticsMainBarChartElements";
-import { displayMonths, allOptions as newOptions, allDatasets as newDatasets } from "./AnalyticsUtils";
 import { useAnalyticsChartCustomHook } from "./useAnalyticsChartCustomHook";
 import { Chart as ChartJS, CategoryScale, LinearScale, BarElement, Title, Tooltip, Legend } from "chart.js";
 
+import { options, datasets, xDataSet } from "./MainBarChartDataMonths.json";
+import {
+    options as hoursOptions,
+    datasets as hoursDatasets,
+    xDataSet as xDataSetHours,
+} from "./MainBarChartDataHours.json";
+
 ChartJS.register(CategoryScale, LinearScale, BarElement, Title, Tooltip, Legend);
 
-export function AnalyticsMainBarChart() {
-    const [chartData, chartOptions] = useAnalyticsChartCustomHook({
-        displayMonths: displayMonths(),
-        newDatasets: newDatasets.slice(-1),
-        newOptions: newOptions.slice(0, 1),
+export function AnalyticsMainBarChart({ timePeriod }) {
+    const [chartValues, setChartValues] = useState({
+        newOptions: options,
+        newDatasets: datasets,
+        displayMonths: xDataSet,
     });
+    useEffect(() => {
+        const newChartValues =
+            timePeriod === "hours"
+                ? { newOptions: options, newDatasets: datasets, displayMonths: xDataSet }
+                : { newOptions: hoursOptions, newDatasets: hoursDatasets, displayMonths: xDataSetHours };
+        setChartValues(newChartValues);
+    }, [timePeriod]);
+
+    const [chartData, chartOptions] = useAnalyticsChartCustomHook({ ...chartValues, timePeriod });
```

Comment:
Hmm, it looks like you don't use the custom hook properly. I think you can modify the hours and months options inside the custom hook. Did you try it?